src/main/java/seedu/jimi/model/FilteredListManager.java

```diff
@@ -71,6 +80,41 @@ public FilteredListManager(TaskBook taskBook) {
     }
     
     /**
+     * @@author A0138915X
+     * 
+     * Wraps all the filteredLists with sortedLists.
+     */
+    private void initSortedLists() {
+        for(ListId id : ListId.values()) {
+            SortedList<ReadOnlyTask> sortedList = new SortedList<ReadOnlyTask>(listMap.get(id));
+            sortedList.setComparator(new Comparator<ReadOnlyTask>() {
+                @Override
+                public int compare(ReadOnlyTask arg0, ReadOnlyTask arg1) {
+                    if(arg0 instanceof Event) {
+                        if(arg1 instanceof Event) {
+                            return ((Event) arg0).getStart().compareTo(((Event) arg1).getStart());
+                        } else { //return -2, lowest natural ordering
+                            return -2;
+                        }
+                    } else if(arg0 instanceof DeadlineTask) {
+                        if(arg1 instanceof DeadlineTask) {
+                            return ((DeadlineTask) arg0).getDeadline().compareTo(((DeadlineTask) arg1).getDeadline());
+                        } else { //return -1, 2nd lowest natural ordering
+                            return -1;
+                        }
+                    } else {
+                        //compare names of floating tasks
+                        return arg0.getName().fullName.compareToIgnoreCase(arg1.getName().fullName);
+                    }
```

Comment:
that arrow head code doe, but im not sure how to avoid in this case since its an inline instance declaration.